components/layout/LanguageSelector.js

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useContext, useEffect } from 'react';
+import PageContext from '../../contexts/PageContext';
+import Router, { useRouter } from 'next/router';
+
+const LanguageSelector = () => {
+    const router = useRouter();
+    const [language, setLanguage] = useState('');
+    const { languageProps } = useContext(PageContext);
+
+    useEffect(() => {
+        setLanguage(languageProps.selectedLanguage);
+    }, []);
+
+    const handleChange = (value) => {
+        localStorage.setItem('dotcms_language', value);
+        setLanguage(value);
+        const {
+            query: { slug }
+        } = router;
+
+        const newRoute =
+            value === process.env.NEXT_PUBLIC_DEFAULT_LANGUAGE
```

Comment:
We need to move this to a function with a more self explainatory name: https://github.com/ryanmcdermott/clean-code-javascript#function-names-should-say-what-they-do